pallets/migrations/src/lib.rs

```diff
@@ -217,14 +217,17 @@ pub mod pallet {
 
 	#[pallet::genesis_config]
 	#[derive(Default)]
-	pub struct GenesisConfig {
-		pub completed_migrations: Vec<Vec<u8>>,
-	}
```

Comment:
I'm fine with leaving this in case you want to allow the user to manually do something other than the default. But I'm also fine with removing it since that probably won't be a very common use case.

Reply:
I'm fine either way, also. It would be trivial to add this back in when needed, although it would require code change (whereas a genesis can be changed without touching code).

Reply:
One good reason to leave it in is that we may want to list migrations which we have deprecated and removed from code.

Reply:
> One good reason to leave it in is that we may want to list migrations which we have deprecated and removed from code.

I don't understand. If a migration has been deprecated and removed from the code, why would we need to list it in a new chain spec?
From my side I don't see any case, so I maintain that we have to delete this useless configuration.
If you see a case, please give a concrete example.

Reply:
Here are two. One concrete, the other more abstract.
1. If you have this field you can use it in your ext builder so that each test has a different Genesis. That way we can make sure skipping is handled correctly in tests.
2. We brainstormed in the future that migrations might specify a dependency graph. In that case we may need to record some prehistoric migrations so that the dependencies of a later migration are satisfied.

Reply:
1. is not a valid case. We can add a parameter to the ext builder to test different storages in the genesis without exposing any configuration in the chain spec. In fact we have to do it, because we are not supposed to expose a feature that is only used for testing.

Reply:
2. At the moment we don't do that, and we may never do. It seems to me that it is a bad practice to maintain a feature for a hypothetical need that we may never have.
We'll hand over this feature the day we'll need it (if that day comes).

Reply:
I agree with you on point 2. I don't really understand what you mean about point 1 though. I'm thinking of a pattern like this https://github.com/PureStake/moonbeam/blob/master/pallets/author-mapping/src/mock.rs#L167-L171 How can you do that without the genesis config?

Anyway, I'm fine with this PR going in as it is if you prefer it this way.

Reply:
Oh, I see what you mean. I just saw your `with_uncompleted_migrations` function. I like that better than what I suggested.